mne/chpi.py

```diff
@@ -366,9 +378,180 @@ def _time_prefix(fit_time):
 
 
 @verbose
+def _fit_cHPI_amplitudes(raw, time_sl, hpi, fit_time, verbose=None):
+    """Fit amplitudes for each channel from each of the N cHPI sinusoids.
+
+    Returns
+    -------
+    sin_fit : ndarray, shape (n_freqs, n_channels)) or None :
+        The sin amplitudes matching each cHPI frequency
+            or None if this time window should be skipped
+    """
+    with use_log_level(False):
+        # loads good channels
+        meg_chpi_data = raw[hpi['meg_picks'], time_sl][0]
+
+    mchpi_data = np.mean(meg_chpi_data, axis=1, keepdims=True)
+    this_data = meg_chpi_data - mchpi_data
+
+    # which HPI coils to use
+    # other then erroring I don't see this getting used elsewhere?
+    if hpi['hpi_pick'] is not None:
+        with use_log_level(False):
+            # loads hpi_stim channel
+            chpi_data = raw[hpi['hpi_pick'], time_sl][0]
+
+        ons = (np.round(chpi_data).astype(np.int) &
+               hpi['on'][:, np.newaxis]).astype(bool)
+        n_on = np.sum(ons, axis=0)
+        if not (n_on >= 3).all():
+            logger.info(_time_prefix(fit_time) + '%s < 3 HPI coils turned on, '
+                        'skipping fit' % (n_on.min(),))
+            return None
+        # #TODO REMOVE # ons = ons.all(axis=1)  # which HPI coils to use
+
+    n_freqs = hpi['n_freqs']
+    this_len = time_sl.stop - time_sl.start
+    if this_len == hpi['n_window']:
+        model, inv_model = hpi['model'], hpi['inv_model']
+    else:  # first or last window
+        model = hpi['model'][:this_len]
+        inv_model = linalg.pinv(model)
+    X = np.dot(inv_model, this_data.T)
+    X_sin, X_cos = X[:n_freqs], X[n_freqs:2 * n_freqs]
+
+    # use SVD across all sensors to estimate the sinusoid phase
+    sin_fit = np.zeros((n_freqs, X_sin.shape[1]))
+    for fi in range(n_freqs):
+        u, s, vt = np.linalg.svd(np.vstack((X_sin[fi, :], X_cos[fi, :])),
+                                 full_matrices=False)
+        # the first component holds the predominant phase direction
+        # (so ignore the second, effectively doing s[1] = 0):
+        X[[fi, fi + n_freqs], :] = np.outer(u[:, 0] * s[0], vt[0])
+        sin_fit[fi, :] = vt[0]
+
+    data_diff = np.dot(model, X).T - this_data
+
+    # compute amplitude correlation (For logging?)
+    g_sin = 1 - np.sqrt((data_diff**2).sum() / (this_data**2).sum())
+    g_chan = 1 - np.sqrt((data_diff**2).sum(axis=1) /
+                         (this_data**2).sum(axis=1))
+    logger.debug('    HPI amplitude correlation %0.3f: %0.3f '
+                 '(%s chnls > 0.90)' % (fit_time, g_sin,
+                                        (g_chan > 0.90).sum()))
+
+    return sin_fit
+
+
+@verbose
+def compute_head_localization(raw, initalLocs=None, time_win=[0, 1],
```

Comment:
I'll move it to private.

I think that there are numerous use cases to want to recompute a dev to head transform. Say I want to use the mean head location for the entire run or maybe I have 4 tasks in a run but am only localizing one and want an optimal head loc for that trial?